pkg/services/alerting/notifiers/slack.go

```diff
@@ -359,9 +359,14 @@ func (sn *SlackNotifier) Notify(evalContext *alerting.EvalContext) error {
 	if evalContext.Rule.State != models.AlertStateOK { //don't add message when going back to alert state ok.
 		msg = evalContext.Rule.Message
 	}
+	// detect google gcs signed header, for example: https://storage.googleapis.com/<project>/alerts/<filename>>?X-Goog-Algorithm=<sha>&X-Goog-Credential=<account>%2Fauto%2Fstorage%2Fgoog4_request&X-Goog-Date=<date>&X-Goog-Expires=<expiration-seconds>&X-Goog-Signature=<signature>&X-Goog-SignedHeaders=<headers>
```

Comment:
I don't understand the changes for the slack notifier. What are you trying to achieve with this? This is a specific GCS check in a specific notifier implementation which we cannot accept. I would also expect that there are other notifiers possibly in need of similar change?

Reply:
@marefr there's logic in the slack notifier that basically only sets the public image url if there's no token. otherwise it tries to upload the image, which is not what i'd like. perhaps an option is to add another setting to the slack notifier to use the public url without uploading the image. otherwise the image is not set at all. what do you think?

https://github.com/grafana/grafana/pull/26840/commits/36eeea5723412369dbaf5b0eeddbba293e254b01#diff-df754044d915ad2fffdd454f27102c53R368

Reply:
Yeah, so suppose you're using Slack in Grafana today and you're not using GCS you have to configure a bot token for slack so that the image is being uploaded. Note that slacak message and slack image is seen as two different messages. If you're using GCS today (pubic access to bucket) with Slack you don't want to configure a bot token and end result is one message in Slack. With your changes I don't see any reason for changing the slack notifier  the end result is the same, a URL either unsigned or signed. What am I missing here?

Reply:
what you're missing is that I tried that path and it didn't work. see https://github.com/grafana/grafana/pull/26840#issuecomment-670642490. the issue is that because there is a token, the image url was not set.

Reply:
Okay you did use a slack bot token. That explains why it didn't work. 

> i could move that to another PR and just leave this one with the gcs uploader changes. what do you think?

Yes, I definitely think so

Reply:
ok. removed custom logic. created #27229 and will work on that next.

Reply:
@marefr commits pushed and master merged. as soon as this is merged will start working on #27229